### ISSUE TITLE
YOLOv4 image resizing

### ISSUE THREAD

**Issue by nizarmasmoudi** [AlexeyAB/darknet#7349]
I am trying to understand the concept of resizing input images to fit network size. I have been through several issues in this Github repository and I haven't seen a clear explanation.

I, let's say, have an image of 1920 x 2000 and the network size is 416 x 416, does that mean the input image will be "compressed/squeezed" into a smaller shape. Wouldn't that mean that small objects might be squeezed into almost 1 pixel ?

If I can have the link to the code sample where the squeezing is performed to further understand this concept, that would be appreciated.

Thank you !

**Comment by stephanecharette** [comment]
See:  https://www.ccoderun.ca/programming/darknet_faq/#square_network and https://www.ccoderun.ca/programming/darknet_faq/#time_to_train

If your objects become "almost 1 pixel" when you resize to your network dimensions, then Darknet/YOLO wont find them.

**Comment by nizarmasmoudi** [comment]
> See: https://www.ccoderun.ca/programming/darknet_faq/#square_network and https://www.ccoderun.ca/programming/darknet_faq/#time_to_train
> 
> If your objects become "almost 1 pixel" when you resize to your network dimensions, then Darknet/YOLO wont find them.

Thank you ! This is answers a lot of questions I had 👍🏼

**Comment by rabsym** [comment]
> See: https://www.ccoderun.ca/programming/darknet_faq/#square_network and 

Before reading this I thought yolo keeps aspect ratio, ie. image 1280x720 will be resized to 416x234, and then inserted into 416x416 network.
Maybe I was wrong :?

**Comment by stephanecharette** [comment]
You are wrong.  As described in the link, it does not maintain aspect ratio.

**Comment by TobiasVorsmann** [comment]
If you set letter_box = 1 in [net] section it keeps the aspect ratio (section For training only -> Data augmentation -> letter_box: [https://github.com/AlexeyAB/darknet/wiki/CFG-Parameters-in-the-%5Bnet%5D-section](https://github.com/AlexeyAB/darknet/wiki/CFG-Parameters-in-the-%5Bnet%5D-section))

**Comment by lsd1994** [comment]
Here is the difference between resize and letter_box.
https://github.com/AlexeyAB/darknet/issues/232#issuecomment-336955485
